app/javascript/packs/application.js

```diff
@@ -11,3 +11,18 @@ import "channels"
 Rails.start()
 Turbolinks.start()
 ActiveStorage.start()
+
+document.addEventListener('turbolinks:before-cache', () => {
+  // Manually tear down bootstrap modals before caching. If turbolinks
+  // caches the modal then tries to restore it, it breaks bootstrap's JS.
+  // We can't just use bootstrap's `modal('close')` method because it is async.
+  // Turbolinks will cache the page before it finishes running.
```

Comment:
Have we clarified what caching is / why it's good to do that for CSS?

Reply:
Good catch. Not yet, I plan to discuss Rails caching later, if that's what you mean — if that's not what you mean, I don't know what you're referring to with "for CSS", perhaps you'll have to teach me and them.

This is a Turbolinks fix required to make Bootstrap modals work properly upon navigation with the browser back button.

Reference: https://www.honeybadger.io/blog/turbolinks/

Reply:
Interesting, I don't think I've seen that behavior before. Good to know.

> I don't know what you're referring to with "for CSS"

I think was thinking about the client side web caching, specifically why it's worth caching / not caching CSS files specifically since it seemed like this was related to CSS.

[This definition](https://www.mnot.net/cache_docs/#DEFINITION) breaks down different types of browser caching that I didn't know the details of before. I found it in this [SO answer](https://stackoverflow.com/a/460465/10481804).